src/request.cpp

```diff
@@ -101,16 +125,51 @@ void Request::process(MegaClient* client)
         auto cmdJSON = client->json;
         bool parsedOk = true;
 
-        Error e;
-        if (cmd->checkError(e, client->json))
+        if (*client->json.pos == ',') ++client->json.pos;
+
+        if (cmd->mSeqtagArray && client->json.enterarray())
         {
-            parsedOk = cmd->procresult(Command::Result(Command::CmdError, e));
+            // Some commands need to return not just a seqtag but also some JSON, in which case they are in an array
+            // These can also return 0 instead of a seqtag instead if no actionpacket was produced.  Coding for any error in that field
+            assert(cmd->mV3);
+            if (client->json.isnumeric())
+            {
+                if (error e = error(client->json.getint()))
+                {
+                    cmd->procresult(Command::Result(Command::CmdError, e));
+                    parsedOk = false; // skip any extra json delivered in the array
+                }
+                else
+                {
+                    parsedOk = processCmdJSON(cmd);
+                }
```

Comment:
Yes, I think you are right.  At the time that was written, the checkError call was earlier in this function rather than moved into processCmdJSON.   I'll try that change out with the integration tests.